base-permissions/src/commonMain/kotlin/PermissionManager.kt

```diff
@@ -59,66 +62,34 @@ interface PermissionManager<P : Permission> {
  * @param stateRepo The [PermissionStateRepo] managed by this manager.
  */
 abstract class BasePermissionManager<P : Permission>(
-    override val permission: P,
-    private val settings: Settings,
-    private val coroutineScope: CoroutineScope
+    final override val permission: P,
+    settings: Settings,
+    coroutineScope: CoroutineScope
 ) : PermissionManager<P>, CoroutineScope by coroutineScope {
 
     data class Settings(
-        val logLevel: LogLevel = LogLevel.NONE
+        val logger: Logger = RestrictedLogger(RestrictedLogLevel.None)
     )
 
-    enum class LogLevel {
-        NONE,
-        INFO,
-        VERBOSE
-    }
+    protected val logTag = "PermissionManager $permission"
+    protected val logger = settings.logger
 
-    private val logTag = "PermissionManager $permission"
-    private val sharedEvents = Channel<PermissionManager.Event>(UNLIMITED)
-    override val events: Flow<PermissionManager.Event> = sharedEvents.receiveAsFlow()
+    protected val eventChannel = Channel<PermissionManager.Event>(UNLIMITED)
+    override val events: Flow<PermissionManager.Event> = eventChannel.receiveAsFlow()
 
     override fun requestPermission() {
-        logInfo { "Request Permission" }
+        logger.info(logTag) { "Request Permission" }
```

Comment:
I put some TODO on other methods like this, it's a bit dangerous to override methods in an abstract class that you still mean to have implemented. Additionally the burden is put on the implementation to call super.

A common pattern is to do something like:
```kotlin
final override fun requestPermission() { // 
        logger.info(logTag) { "Request Permission" }
        requestPermissionImplementation()
}
protected abstract fun requestPermissionImplementation()
```

not ideal but IMHO better than what we have here.